keras/optimizers.py

```diff
@@ -69,6 +70,9 @@ def get_updates(self, params, constraints, loss):
 
     def get_gradients(self, loss, params):
         grads = K.gradients(loss, params)
+        if hasattr(self, 'normalize_gradients') and self.normalize_gradients:
+            norm = K.sqrt(sum([K.sum(K.square(g)) for g in grads])) + K.epsilon()
```

Comment:
Rather, we should use the `l2_normalize` backend function for better performance. Note that it requires specifying the axis or axes. I believe the correct approach in general is to normalize along all axes.